### PR TITLE
[collectd 6] Enable `-Wstrict-prototypes`.

### PR DESCRIPTION
**Pull request by octo**
In pre-ANSI K&R-style C, functions can use an "initializer list" instead of a "parameter type list". Compilers still support this for backwards compatibility, unfortunately.

One result is that compilers will accept the following prototype:

```c
void foo();
```

Note that this probabaly does not mean what you think it means: this is a function with *an unknown number of arguments* and compilers will happily let you pass anything to it. In other words, the following would be legal:

```c
foo("Hello, World!");
```

This PR changes all (unintentional) uses of empty initializer lists to use `(void)` instead, which expresses what is intended: "this function takes no arguments". It also adds the `-Wstrict-prototypes` CFLAG to prevent backsliding.

### CODE REVIEW

**Comment by eero-t** [comment]
> Note that this probabaly does not mean what you think it means: this is a function with _an unknown number of arguments_ and compilers will happily let you pass anything to it.

FYI: I filed GCC bug 13 years ago about that not giving a warning (even when large number of generic GCC warning options are enabled), and it's still open: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=48091

**Review by eero-t — APPROVED** [review]
Approved, but both of these failures happen on multiple CI checks:
* configure: error: `strtok_r` is not available. Please file a bugreport!
* expected plugins are MISSING: `bind`

First error happens on older OSes, latter one on newer ones.

PS. Please apply similar compiler option fix also to `main`!

**Comment by eero-t** [comment]
```
src/bind.c:420:15: error: implicit declaration of function 'strptime'; did you mean 'strftime'? [-Werror=implicit-function-declaration]
  420 |   char *tmp = strptime(str_ptr, "%Y-%m-%dT%T", &tm);
      |               ^~~~~~~~
      |               strftime
src/bind.c:420:15: error: initialization of 'char *' from 'int' makes pointer from integer without a cast [-Werror=int-conversion]
```

**Comment by octo** [comment]
The problem is that the configure script relies heavily on K&R style functions, for example:

```
configure:13281: checking for dlopen
configure:13281: gcc -o conftest -O0 -g -Wstrict-prototypes   conftest.c  >&5
conftest.c:63:1: warning: function declaration isn't a prototype [-Wstrict-prototypes]
   63 | char dlopen ();
      | ^~~~
configure:13281: $? = 0
configure:13281: result: yes
```

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -64,12 +68,14 @@ jobs:
       with:
         name: ${{ matrix.container_tag }}
         path: config.log
+    # The configure script relies on the old syntax to check whether functions
+    # exist, so we can't specify `-Wstrict-prototypes` globally.
```
Comment (eero-t) [inline .github/workflows/build.yml:72]: Maybe better place for this comment is in the configure step:
```suggestion
    # Configure scripts rely on the K&R syntax to check whether functions
    # exist, so `-Wstrict-prototypes` can be specified only for make.
```

**Comment by eero-t** [comment]
> It is absolutely incomprehensible to me why `-Wall` doesn't include `-Wstrict-prototypes`.

Especially when it's not enabled even if one adds e.g. `-Wextra -Wpedantic -std=c99`...

**Pull request merged**